### ISSUE TITLE
[used-before-assignment] False negative for for/if/else/continue pattern

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#9379]
### Bug description

```python
def halfway_safe():
    """Name used safely inside the loop, unsafely outside it."""
    for _temp in range(0, 1):
        try:
            pass
        except ValueError:
            error = True
        else:
            continue

        print(error)
    print(error)  # Should raise used-before-assignment
```

### Command used

```shell
n/a
```

### Pylint output

```python
nothing
```

### Expected behavior

used-before-assignment on last line

### Pylint version

```shell
#9374
```
